table.go

```diff
@@ -388,7 +391,7 @@ func (t *genTable[Obj]) ListWatch(txn ReadTxn, q Query[Obj]) (iter.Seq2[Obj, Rev
 		// Doing a Get() is more efficient than constructing an iterator.
 		value, watch, ok := indexTxn.Get(q.key)
 		seq := func(yield func(Obj, Revision) bool) {
-			if ok {
+			if ok && !value.deleted {
 				yield(value.data.(Obj), value.revision)
 			}
 		}
```

Comment:
Also in the CompareAndSwap impl; should the compare not first check that the compared to obj is not deleted?

Reply:
Not sure what you mean. We can't look into `value` unless `ok` is true.

Reply:
uhh, I think I was trying to comment on CompareAndDelete below but couldn't because GitHub UI

but the inner `delete` handles the `deleted` objects so I think this is just fine